vsphere_plugin_common/__init__.py

```diff
@@ -1498,6 +1518,24 @@ def create_server(
                 guest_map.adapter.subnetMask = str(nw.netmask)
                 adaptermaps.append(guest_map)
 
+        # attach cdrom
+        if ide_controller and cdrom_image:
```

Comment:
Does this mean that the CD image will silently fail to mount if there's no ide controller? (actually I think it'll enter an endless retry loop because ide_controller won't be set))

Reply:
I will add raise exception if we don't have ide free controller.

Reply:
OK- that's not something that can be dynamically fixed (by attaching one)?